### PR TITLE
[TS] Migrate <Grommet/> tests to .tsx 

### PR DESCRIPTION
**Pull request by Ojas-Sri**
#### What does this PR do?
- Migrates `Grommet-test` to Typescript under #5599 
- Could not find type declarations for `MessageContext` , nor could infer the same from its implementation. Therefore, I have commented out those tests for now as `<Grommet>` component now supports different type of `messages` object

#### Where should the reviewer start?
- `Grommet-test.tsx` file (was renamed from `Grommet-test.js`)
#### What testing has been done on this PR?
- Unit Testing 
#### How should this be manually tested?
- N/A
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
#5599 
#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Looks like it

### CODE REVIEW

File: src/js/components/Grommet/__tests__/Grommet-test.tsx [hunk 1]
```diff
@@ -85,39 +88,30 @@ describe('Grommet', () => {
...
+  //         },
+  //       }}
+  //     >
+  //       <MessageContext.Consumer>
+  //         {({ format }) => format({ id: 'test.label' })}
+  //       </MessageContext.Consumer>
+  //     </Grommet>,
+  //   );
+  //   expect(container.firstChild).toMatchSnapshot();
+  // });
```
Comment (jcfilben) [inline src/js/components/Grommet/__tests__/Grommet-test.tsx:108]:
```suggestion
  test('messages', () => {
    const { container } = render(
      <Grommet
      messages={{
        messages: {
            // @ts-ignore
            test: {
              label: 'My Label',
            },
          },
        }}
      >
        <MessageContext.Consumer>
          {({ format }) => format({ id: 'test.label' })}
        </MessageContext.Consumer>
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
```

Can we leave this test in and just add `// @ts-ignore`? The types for the Grommet `messages` prop are pretty out of date and need updated but I think that can be handled separately.

File: src/js/components/Grommet/__tests__/Grommet-test.tsx [hunk 2]
```diff
@@ -85,39 +88,30 @@ describe('Grommet', () => {
...
     const { container } = render(
       <Grommet full background="#0000ff">
         Grommet App
       </Grommet>,
-
-      <Grommet messages={{ format: (opts) => messages[opts.id] }}>
-        <MessageContext.Consumer>
-          {({ format }) => format({ id: 'test.label' })}
-        </MessageContext.Consumer>
-      </Grommet>,
```
Comment (jcfilben) [inline src/js/components/Grommet/__tests__/Grommet-test.tsx:120]: Same comment here, can we leave this in and use `// @ts-ignore`?

**Review by jcfilben — COMMENTED** [review]
Thanks for working on this! Just a few small comments

File: src/js/components/Grommet/__tests__/Grommet-test.tsx (hunk 1 above)
Comment (Ojas-Sri) [reply src/js/components/Grommet/__tests__/Grommet-test.tsx:108]: sure

**Pull request closed without merging**